for_developers/product_design.md

```diff
@@ -0,0 +1,114 @@
+   These special functions, namely `create_model()`, `customize_inputs()` and `customize_outputs()`, facilitate the conversion of our own data entity class to the format required by a model imported from an external framework.
+
+   Let's consider an example of importing a model from MMDetection[^4]:
+
+   | ![Reuse Model](./images/product_design/reuse_model.png) |
+   | :-----------------------------------------------------: |
+   | Figure 2. Import RTMDetTiny[^5] model from MMDetection  |
+
+   In Figure 2, `MMDetCompatibleModel` implements `create_model()` to generate the `RTMDetTiny` model and registers it as a child module.
+   At this point, the `RTMDetTiny` model requires `DataSample` input from MMDetection and produces a Python dictionary for its output.
+   These data entity types do not comply with our framework.
+   To resolve this issue, `MMDetCompatibleModel` implements `customize_inputs()` and `customize_outputs()` functions to convert them into our format.
+   With this design, we can explicitly support major frameworks. We plan to continually expand the supported frameworks using this approach.
+
+3. **Support Various Data Formats without Worrying about Customizing the Input Data Pipeline**
+
+   _Much of the time for an AI project is likely to be spent on data-related tasks ..._[^6]
+
+   When developing a new deep learning model, one of the most tedious tasks is often creating a data pipeline for the given dataset.
+   Developers must study the data schema of the provided dataset and parse it into a data entity consumable by both the data augmentation pipeline and the model.
+
+   To simplify and enhance this process in the model development lifecycle, we introduce Datumaro[^7], a member of the OpenVINO™ open-source ecosystem.
+   Datumaro allows the import of various data formats from around the world.
+   Please refer to the following Figure.
+
+   | ![Support Various Data Format](./images/product_design/support_various_data_format.png) |
+   | :-------------------------------------------------------------------------------------: |
+   | Figure 3. Only the root directory path is provided by the user to make a data pipeline  |
+
+   In Figure 3, a popular data format such as COCO, VOC, YOLO, or Cityscape can be imported with Datumaro.
+   While Figure 3 illustrates the entire data pipeline, the only input required from the user is `data_root`.
+   This simplicity allows users to enjoy a convenient model training workflow without worrying about the input data pipeline.
+
+   Another advantage highlighted in Figure 3 is the support for multiple data augmentation pipelines.
+   The default recommendation is TorchVision[^8].
+   However, as shown, there are other pipelines such as MMPretrain, MMDetection, etc.
+   This flexibility enables developers to freely compose the model and the data augmentation pipeline from various frameworks, accelerating the creation of custom model training templates for specific problem domains.
```

Comment:
Sorry that I felt these are not sufficient. Why don't you adding some backgrounds of introduction to factory type transformlib to support various customized augmentations per frameworks?

Reply:
Updated in https://github.com/openvinotoolkit/training_extensions/pull/2741/commits/3051270768fca55c33a0009368687c6a2cf80fb3